### ISSUE TITLE
Change Releases warning Message to "will require >PHP8.0  or <PHP8.2 "

### ISSUE THREAD

**Issue by nivadis** [grokability/snipe-it#12357]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

please change the messeage of the releases to >PHP8.0  or <PHP8.2 and not PHP8.0 or greater

the cli with php8.2 gave the hint to avoid php8.2  ;-)

thanks

### Reproduction steps

1.
2.
3.
...

### Expected behavior

>PHP8.0  or <PHP8.2 and not PHP8.0 or greater

### Snipe-IT Version

v6.0.14 build 9161 (g799c9c910)

### Operating System

debian

### Web Server

apache

### PHP Version

8.2.X

**Comment by snipe** [comment]
What message are you getting with PHP8.2? It *should* work with 8.2 (*should* being the operative word here.)

**Comment by uberbrady** [comment]
`composer install` does not work on 8.2 right now, I can confirm that on my local. Let me see if I can twiddle some stuff.

**Comment by p-rintz** [comment]
Without running "composer update", the following will error out:

```
Installing dependencies from lock file
Verifying lock file contents can be installed on current platform.
Your lock file does not contain a compatible set of packages. Please run composer update.

  Problem 1
    - nette/schema is locked to version v1.2.2 and an update of this package was not requested.
    - nette/schema v1.2.2 requires php >=7.1 <8.2 -> your php version (8.2.2) does not satisfy that requirement.
  Problem 2
    - nette/utils is locked to version v3.2.7 and an update of this package was not requested.
    - nette/utils v3.2.7 requires php >=7.2 <8.2 -> your php version (8.2.2) does not satisfy that requirement.
  Problem 3
    - phpspec/prophecy is locked to version v1.15.0 and an update of this package was not requested.
    - phpspec/prophecy v1.15.0 requires php ^7.2 || ~8.0, <8.2 -> your php version (8.2.2) does not satisfy that requirement.
  Problem 4
    - nette/schema v1.2.2 requires php >=7.1 <8.2 -> your php version (8.2.2) does not satisfy that requirement.
    - league/config v1.1.1 requires nette/schema ^1.2 -> satisfiable by nette/schema[v1.2.2].
    - league/config is locked to version v1.1.1 and an update of this package was not requested.
```

**Comment by KorvinSzanto** [comment]
@snipe is the fix at this point to run `composer update` with php8.2 or to downgrade to PHP8.1?

**Comment by carbon-evolution** [comment]
Is the issue resolved?
Getting same error as of @p-rintz.

PHP 8.2.8 (cli) (built: Jul  8 2023 07:10:21) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.8, Copyright (c) Zend Technologies
    with Zend OPcache v8.2.8, Copyright (c), by Zend Technologies

  Problem 1
    - Root composer.json requires php >=7.4.3 <8.2 but your php version (8.2.8) does not satisfy that requirement.
  Problem 2
    - nette/schema is locked to version v1.2.2 and an update of this package was not requested.
    - nette/schema v1.2.2 requires php >=7.1 <8.2 -> your php version (8.2.8) does not satisfy that requirement.
  Problem 3
    - nette/utils is locked to version v3.2.7 and an update of this package was not requested.
    - nette/utils v3.2.7 requires php >=7.2 <8.2 -> your php version (8.2.8) does not satisfy that requirement.
  Problem 4
    - phpspec/prophecy is locked to version v1.15.0 and an update of this package was not requested.
    - phpspec/prophecy v1.15.0 requires php ^7.2 || ~8.0, <8.2 -> your php version (8.2.8) does not satisfy that requirement.
  Problem 5
    - nette/schema v1.2.2 requires php >=7.1 <8.2 -> your php version (8.2.8) does not satisfy that requirement.
    - league/config v1.1.1 requires nette/schema ^1.2 -> satisfiable by nette/schema[v1.2.2].
    - league/config is locked to version v1.1.1 and an update of this package was not requested.

**Comment by uberbrady** [comment]
Your best bet is to downgrade to php8.1

**Comment by KiralyCraft** [comment]
I'm running PHP 8.2.10 on Arch Linux. I've modified composer.json to edit the restrictions for < 8.2.11 instead of < 8.2, ran the composer command with "update" instead of "install ....", and then installed as indicated (this may be redundant though).

So far it works wonders, I haven't found any issues with it.

**Comment by CharmingRhino** [comment]
> I'm running PHP 8.2.10 on Arch Linux. I've modified composer.json to edit the restrictions for < 8.2.11 instead of < 8.2, ran the composer command with "update" instead of "install ....", and then installed as indicated (this may be redundant though).
> 
> So far it works wonders, I haven't found any issues with it.

Which lines in the composer.json file did you have to change?

**Comment by KiralyCraft** [comment]
> Which lines in the composer.json file did you have to change?

[This](https://github.com/snipe/snipe-it/blob/16498fdcf82f76d3f9e992abec56e769fab6c205/composer.json#L20) one

**Comment by nemchik** [comment]
Just chiming in; It was pointed out above that this line https://github.com/snipe/snipe-it/blob/3a6d626aa13fb0399100554f44ba424c3e2cbc53/composer.json#L20 prevents using php 8.2 (and now also php 8.3) but actually still allows using php 7.4 (EOL) and php 8.0 (EOL).

Per https://www.php.net/supported-versions.php as of writing, php 8.1 (the only version of php that is not EOL that works to install snipe-it) is no longer in active support. It is still receiving security updates and is not yet EOL, but it will be as of 25 Nov 2024. The two versions of php currently in active support are php 8.2 and php 8.3.

The composer.json file should be updated to reflect the supported versions of php if php 8.2 is meant to be included. Whether EOL versions of php are supported is up to the project to decide, but if the only versions of php that a project supports are outside of php's active support window it paints a bit of a negative picture.

@snipe an update to the composer.json would be greatly appreciated if the supported php version range has changed.

As a side note, I am also involved in maintaining https://github.com/linuxserver/docker-snipe-it and have been unable to make a few changes due to being locked in with the older (now EOL) php 8.1.

<blockquote><img src="https://www.php.net/favicon.ico?v=2" width="48" align="right"><div><strong><a href="https://www.php.net/supported-versions.php">PHP: Supported Versions</a></strong></div></blockquote>
<blockquote><img src="https://opengraph.githubassets.com/4cab9de4fce99bc2bbf4435e83a3f702d5976595d5c64c9cff432d18b7810894/linuxserver/docker-snipe-it" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/linuxserver/docker-snipe-it">GitHub - linuxserver/docker-snipe-it: Alpine/Nginx container for the Asset Management software Snipe-IT</a></strong></div><div>Alpine/Nginx container for the Asset Management software Snipe-IT - GitHub - linuxserver/docker-snipe-it: Alpine/Nginx container for the Asset Management software Snipe-IT</div></blockquote>

**Comment by snipe** [comment]
@nemchik Supported versions have not been changed. Snipe-IT v7 will phase out 7.4, but we're stuck with supporting it until that comes out.

**Comment by vithurshanselvarajah** [comment]
@snipe  

Im currently using php 8.2.10 and have to perform the upgrades manually which isn't an issue in the short term. 

Will Snipe-IT v7 introduce support to be able to upgrade with this PHP version? I'm questioning if life would be made easier downgrading PHP or if it makes more sense to just wait. 

The most recent version of upgrade.php introduces a a requirements JSON file which only supports 8.2.0 any newer minor versions aren't supported.

**Comment by snipe** [comment]
@vithusel Yeah, that new file is to make upgrading easier moving forward. Since we don’t know what version someone is upgrading from/to, this at least gives us a way to stop people from blowing up their installations if they missed the PHP requirements (both max and min) in the documentation. We’re still refining that file though, since it’s not actually used yet. 

8.2 isn’t supported in the Snipe-IT v6 line (due to dependencies, not because we don’t *want* to), but v7 will be.

**Comment by vithurshanselvarajah** [comment]
> @vithusel Yeah, that new file is to make upgrading easier moving forward. Since we don’t know what version someone is upgrading from/to, this at least gives us a way to stop people from blowing up their installations if they missed the PHP requirements (both max and min) in the documentation. We’re still refining that file though, since it’s not actually used yet. 
> 
> 8.2 isn’t supported in the Snipe-IT v6 line (due to dependencies, not because we don’t *want* to), but v7 will be.

Thank you for the confirmation. 8.2.10 seems to be playing well for me so perhaps I'm narrowly avoiding a particular feature that's broken with it. I'll freeze my servers for now and will wait till v7 to upgrade

**Comment by pvsaintpe** [comment]
<img width="1250" alt="image" src="https://github.com/snipe/snipe-it/assets/16900171/d1465d7f-c6ec-4191-a7ca-48ce66df583b">

**Comment by pvsaintpe** [comment]
good night, can you do something about this?
Laravel 11 works with php8.2/8.3 why can't I install dependencies for php 8.2/8.3 and Laravel 11?

**Comment by snipe** [comment]
@pvsaintpe - per the requirements documentation, we do not yet support PHP 8.2. We will in Snipe-IT v7 which should be out soon.
